задание спринта 7

NastyaDanilova1 opened the pull request:
выполнила задание спринта 7

Commit:
init develop

AlexKitIt commented on conftest.py:
```diff
@@ -0,0 +1,29 @@
+from for_help import Help
+
+
+@pytest.fixture(scope='function')
+def helpers():
+    return Help()
```
Нужно исправить: фикстуры выполняют сложную логику предусловий\постусловий а не создают объект

AlexKitIt commented on for_help.py:
```diff
@@ -0,0 +1,54 @@
+            "firstName": first_name
+        }
+
+        response = requests.post('https://qa-scooter.praktikum-services.ru/api/v1/courier', data=payload)
```
Нужно исправить: урлы у нас уже вынесены

AlexKitIt commented on tests/test_delete_courier.py:
```diff
@@ -0,0 +1,42 @@
+class TestDeleteCourier:
+    @allure.title('проверка удаления курьера со всеми обязательными полями')
+    def test_delete_courier(self, helpers):
+        data = helpers.register_new_courier_and_return_login_password()
+        response_post = requests.post(Const.LOGIN_COURIER, data={
+            "login": data[0],
+            "password": data[1],
+        })
```
Нужно исправить: здесь и далее: эта логика описана в фикстурах, необходимы их вызывать а не копипастить в тесте

Commit:
fix

Commit:
clean imports

Commit:
sprint 7 changes done
- исправлены ошибки
- более лаконично переписан метод создания заказа
- убран класс Help
- убран файл Const, данные перенесены в data

AlexKitIt commented on data.py:
```diff
@@ -0,0 +1,52 @@
+    }
+
+
+class Url:
+    MAIN_URL = 'http://qa-scooter.praktikum-services.ru'
+    CREATE_COURIER = f'{MAIN_URL}/api/v1/courier'
+    CREATE_ORDER = f'{MAIN_URL}/api/v1/orders'
+    ORDER_LIST = f'{MAIN_URL}/api/v1/orders'
+    DELETE_COURIER = f'{MAIN_URL}/api/v1/courier/'
+    GET_ORDER_TRACK = f'{MAIN_URL}/api/v1/orders/track'
+    TAKE_ORDER = f'{MAIN_URL}/api/v1/orders/accept/'
+    LOGIN_COURIER = f'{MAIN_URL}/api/v1/courier/login'
```
Нужно исправить: здесь и далее:  это уже содержится в const , зачем копипаста?

NastyaDanilova1 replied:
Я перенесла данные в файл date и в связи с этим переписала тесты без обращения к файлу const, обращаюсь к data.
Не затушила это изменение в прошлый раз, сейчас исправила

AlexKitIt commented on tests/test_create_order.py:
```diff
@@ -0,0 +1,16 @@
+
+
+class TestCreateOrder:
+    @allure.step("создание заказа с разными цветами самоката")
```
Нужно исправить: над тестами везде стоит title, почему тут step?

AlexKitIt commented on conftest.py:
```diff
@@ -0,0 +1,25 @@
+from for_help import *
+
+
+@pytest.fixture(scope='function')
+def create_order():
+    response_create_order = requests.post(Url.CREATE_ORDER, json=UserData.person_data)
+    track = response_create_order.json()['track']
+    return track
+
```
Нужно исправить: фикстура нигде не используется

Commit:
sprint 7 changes done
- исправлены ошибки
- убран файл Const, данные перенесены в data
